src/goto-instrument/cover.cpp

```diff
@@ -28,34 +29,22 @@ Date: May 2016
 void instrument_cover_goals(
   goto_programt &goto_program,
   const cover_instrumenterst &instrumenters,
-  message_handlert &message_handler)
-{
-  cover_basic_blockst basic_blocks(goto_program);
```

Comment:
The only reason why the basic blocks are computed here is because the location and the branch instrumenter could use it and so we want to avoid recomputation. One could do a more radical refactoring and try to hide the basic block stuff inside the instrumenter, but I'd defer this to a future refactoring.